src/crafting.cpp

```diff
@@ -108,9 +110,55 @@ float player::morale_crafting_speed_multiplier( const recipe &rec ) const
     return 1.0f / morale_effect;
 }
 
-float player::crafting_speed_multiplier( const recipe &rec, bool in_progress ) const
+static float workbench_crafting_speed_multiplier( const tripoint &loc )
 {
-    float result = morale_crafting_speed_multiplier( rec ) * lighting_craft_speed_multiplier( rec );
+    // tripoint_zero indicates crafting from inventory, so no effect
+    if( loc == tripoint_zero ) {
+        return 1.0f;
+    }
+
+    int workbench_level = 0;
```

Comment:
This variable probably isn't needed. You can  return values right after the checks.